dnf-behave-tests/features/repoquery/deps.feature

```diff
@@ -0,0 +1,565 @@
+Feature: Tests for the basic repoquery dependencies functionality:
+    --requires, --provides, --conflicts, --obsoletes
+    --whatrequires, --whatprovides, --whatconflicts, --whatobsoletes
+
+Background:
+ Given I use the repository "repoquery-deps"
+
+
+# --requires
+Scenario: repoquery --requires NAME
+ When I execute dnf with args "repoquery --requires middle1"
+ Then the exit code is 0
+  And stdout is
+      """
+      /a/bottom4-file
+      bottom1-prov1
+      bottom1-prov2 = 1.0
+      bottom1-prov2 >= 2.0
+      bottom2 = 1:1.0-1
+      bottom3
+      """
+
+Scenario: repoquery --requires NAME-VERSION
+ When I execute dnf with args "repoquery --requires middle1-1.0"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1-prov1
+      bottom1-prov2 = 1.0
+      bottom2 = 1:1.0-1
+      bottom3
+      """
+
+Scenario: repoquery --requires --resolve NAME
+ When I execute dnf with args "repoquery --requires --resolve middle1"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1-1:2.0-1.x86_64
+      bottom2-1:1.0-1.x86_64
+      bottom3-1:2.0-1.x86_64
+      bottom4-1:1.0-1.x86_64
+      """
+
+Scenario: repoquery --requires --resolve NAMEGLOB
+ When I execute dnf with args "repoquery --requires --resolve middle[1]"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1-1:2.0-1.x86_64
+      bottom2-1:1.0-1.x86_64
+      bottom3-1:2.0-1.x86_64
+      bottom4-1:1.0-1.x86_64
+      """
+
+Scenario: repoquery --requires --resolve --recursive NAME
+ When I execute dnf with args "repoquery --requires --resolve --recursive top1"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1-1:2.0-1.x86_64
+      bottom2-1:1.0-1.x86_64
+      bottom3-1:2.0-1.x86_64
+      bottom4-1:1.0-1.x86_64
+      middle1-1:2.0-1.x86_64
+      middle2-1:2.0-1.x86_64
+      """
+
+# TODO lists bottom3, while it shouldn't; see the test below with --tree, which is corectly not listing bottom3
+@xfail
+Scenario: repoquery --requires --resolve --recursive NAME-VERSION
+ When I execute dnf with args "repoquery --requires --resolve --recursive top1-2.0"
+ Then the exit code is 0
+  And stdout is
+      """
+      bottom1-1:2.0-1.x86_64
+      bottom2-1:1.0-1.x86_64
+      middle1-1:2.0-1.x86_64
+      middle2-1:2.0-1.x86_64
+      """
```

Comment:
middle1-1:2.0-1.x86_64 which is in the transaction requires /a/bottom4-file
the file is provided bottom4-1:1.0-1.x86_64
I believe it should be added to the output.

bottom3-1:2.0-1.x86_64 is probably pulled it because it obsoletes 'bottom2 = 1:2.0-1'
Probably also expected as the obsolete is in place.

Reply:
> middle1-1:2.0-1.x86_64 which is in the transaction requires /a/bottom4-file
> the file is provided bottom4-1:1.0-1.x86_64
> I believe it should be added to the output.

Yes, that was added later...

> bottom3-1:2.0-1.x86_64 is probably pulled it because it obsoletes 'bottom2 = 1:2.0-1'
> Probably also expected as the obsolete is in place.

Ok, (pulling in obsoletes might be a thing to write a test case for) but I don't see anything depending on `bottom2 = 1:2.0-1`?